src/main/java/com/salesforce/storm/spout/dynamic/kafka/Consumer.java

```diff
@@ -384,6 +385,23 @@ private void reportStatus() {
             );
         }
 
+        // Capture the difference, or the "lag" for this topic/partition on the kafka topic
+        for (final ConsumerPartition consumerPartition : getCurrentState().getConsumerPartitions()) {
+            final TopicPartition topicPartition = new TopicPartition(consumerPartition.namespace(), consumerPartition.partition());
+
+            if (!topicPartitions.containsKey(topicPartition)) {
+                continue;
+            }
+
+            final Long endOffset = topicPartitions.get(topicPartition);
+
+            metricsRecorder.assignValue(
+                getClass(),
+                "lag.topic." + consumerPartition.namespace() + ".partition." + consumerPartition.partition(),
+                endOffset - getCurrentState().getOffsetForNamespaceAndPartition(consumerPartition)
```

Comment:
It'd probably be better to call getCurrentState() once and re-use the instance instead of rebuilding it for every iteration of the loop